### PR TITLE
fix: serve directory without trailing slash fixes(#272)

### LINKED ISSUE #272: enable fastify-static to serve a gatsby site without trailing slashes without redirect
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

I have a gatsby site that doesn't use trailing slashes and I wish to serve it with fastify-static

This means that a request for a path like /page1 should serve the file page1/index.html (or index.html.br with preCompressed)

Currently a 404 is used.

### Motivation

_No response_

### Example

_No response_

### PR DESCRIPTION
**Pull request by cameronbraid**
fixes #272 in conjunction  with https://github.com/fastify/fastify-static/pull/271

### CODE REVIEW

File: test/static.test.js [hunk 1]
```diff
@@ -2537,7 +2537,7 @@ t.test('register /static with redirect true and wildcard false', t => {
   })
 })
 
-t.test('trailing slash behavior with redirect = false', (t) => {
+t.only('trailing slash behavior with redirect = false', (t) => {
```
Comment (mcollina) [inline test/static.test.js:2540]:
```suggestion
t.test('trailing slash behavior with redirect = false', (t) => {
```

File: test/static.test.js [hunk 2]
```diff
@@ -2588,14 +2588,14 @@ t.test('trailing slash behavior with redirect = false', (t) => {
       })
     })
 
-    t.test('deep path with index.html but no trailing slash => 404', (t) => {
+    t.only('deep path with index.html but no trailing slash => 200', (t) => {
```
Comment (mcollina) [inline test/static.test.js:2591]:
```suggestion
    t.test('deep path with index.html but no trailing slash => 200', (t) => {
```

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by mcollina** [comment]
Good work!

File: test/static.test.js (hunk 2 above)
Comment (cameronbraid) [reply test/static.test.js]: whoops !

File: test/static.test.js (hunk 1 above)
Comment (cameronbraid) [reply test/static.test.js]: whoops !

**Pull request merged**